wom/src/test/scala/wdl4s/wdl/wom/WdlScatterWomSpec.scala

```diff
@@ -0,0 +1,203 @@
+package wdl4s.wdl.wom
+
+import lenthall.collections.EnhancedCollections._
+import cats.data.Validated.{Invalid, Valid}
+import org.scalatest.{FlatSpec, Matchers}
+import wdl4s.wdl.types.{WdlArrayType, WdlIntegerType, WdlStringType}
+import wdl4s.wdl.{WdlNamespace, WdlNamespaceWithWorkflow}
+import wdl4s.wom.graph.GraphNodePort.ScatterGathererPort
+import wdl4s.wom.graph._
+
+class WdlScatterWomSpec extends FlatSpec with Matchers {
+
+  behavior of "WdlNamespaces with scatters"
+
+  it should "convert gathered scatter outputs to workflow outputs" in {
+    val scatterTest =
+      """task foo {
+        |  Int i
+        |  command { ... }
+        |  output {
+        |    String out = i
+        |  }
+        |}
+        |
+        |workflow scatter_test {
+        |  Array[Int] xs
+        |  scatter (x in xs) {
+        |    call foo { input: i = x }
+        |  }
+        |}""".stripMargin
+
+    val namespace = WdlNamespace.loadUsingSource(scatterTest, None, None).get.asInstanceOf[WdlNamespaceWithWorkflow]
+    import lenthall.validation.ErrorOr.ShortCircuitingFlatMap
+    val scatterTestGraph = namespace.womExecutable.flatMap(_.graph)
+
+    scatterTestGraph match {
+      case Valid(g) => validateGraph(g)
+      case Invalid(errors) => fail(s"Unable to build wom version of scatter foo from WDL: ${errors.toList.mkString("\n", "\n", "\n")}")
+    }
+
+    def validateGraph(workflowGraph: Graph) = {
+
+      /*
+      Validate that the outer graph is looking good:
+       */
+
+      val scatterNode = workflowGraph.nodes.collectFirst {
+        case s: ScatterNode => s
+      }.getOrElse(fail("Resulting graph did not contain a ScatterNode"))
+
+      val xs_inputNode = workflowGraph.nodes.collectFirst {
+        case gin: GraphInputNode if gin.name == "xs" => gin
```

Comment:
If it had that kind of magic, I wonder could it take an optional predicate function from the type to Boolean?

Reply:
I think once it's that complicated sticking with a `collectFirst` might be easier to quickly understand